src/loki_alert_rules/database_unavailable.rule

```diff
@@ -0,0 +1,7 @@
+alert: DatabaseUnavailable
+expr: rate({%%juju_topology%%} |~ `failed to connect to.*host=postgresql-k8s`[1m]) > 0
```

Comment:
I am new to PromQL. I do have a question about this query. Does it rely on this pattern `failed to connect to blah blah` to detect the error logs? If this is the case, I would not do it this way personally.

And another thing is that this metric does not necessarily shows the database is unavailable. For example, the network issue could also lead to connection issues. IMHO, I would not provide this rule here personally. I think it is more appropriate for the database itself to provide such metrics, e.g. the database could have some alert rule saying how many database units are down, etc.

Reply:
i agree with @wood-push-melon

Reply:
I should have phrased it better, but the alert exists to notify if the hydra can not access the backend, not to be a replacement for alerts for the postgresql backend.